src/go/rpk/pkg/cli/cluster/selftest/start.go

```diff
@@ -143,6 +143,50 @@ func assembleTests(onlyDisk bool, onlyNetwork bool, onlyCloud bool, durationDisk
 			Parallelism: 2,
 			Type:        adminapi.DiskcheckTagIdentifier,
 		},
+		adminapi.DiskcheckParameters{
+			Name:        "4KB sequential r/w, no dsync",
+			DSync:       false,
+			SkipWrite:   false,
+			SkipRead:    true,
+			DataSize:    1 * units.GiB,
+			RequestSize: 4 * units.KiB,
+			DurationMs:  durationDisk,
+			Parallelism: 2,
+			Type:        adminapi.DiskcheckTagIdentifier,
+		},
+		adminapi.DiskcheckParameters{
+			Name:        "16KB sequential r/w, low io depth",
+			DSync:       true,
+			SkipWrite:   false,
+			SkipRead:    true,
+			DataSize:    1 * units.GiB,
+			RequestSize: 16 * units.KiB,
+			DurationMs:  durationDisk,
+			Parallelism: 1,
+			Type:        adminapi.DiskcheckTagIdentifier,
+		},
+		adminapi.DiskcheckParameters{
+			Name:        "16KB sequential r/w, medium io depth",
+			DSync:       true,
+			SkipWrite:   false,
+			SkipRead:    true,
+			DataSize:    1 * units.GiB,
+			RequestSize: 16 * units.KiB,
+			DurationMs:  durationDisk,
+			Parallelism: 8,
+			Type:        adminapi.DiskcheckTagIdentifier,
+		},
+		adminapi.DiskcheckParameters{
+			Name:        "16KB sequential r/w, high io depth",
```

Comment:
Did you intentionally not add something like 4k @ 256 iodepth?

Reply:
Are you asking more about "why not 4K" or "why not 256 iodepth"?

In any case it was intentional but open to ideas here. One thing to note is that the parallelism factor here is then multiplied by the shard count, so on modest 8 shard nodes we are already at a very high 512 io depth, which IME is larger than what you need to get max throughput even on large local SSD configurations (though of course this may not be the case on some _other_ storage configuations, especially high throughput, longer latency network attached storage).

I don't actually like this multiplication because it (a) adds a confounding factor when comparing results against different clusters which may have different shard counts (but at least now we see the effective iodepth in the output) and (b) it means you can't run an iodepth=1 test except on a cluster with 1-shard nodes.

About 4K vs 16K, by goal was to add a 16K test to see the difference between 4K and 16K, i.e., how much performance varies in the range of block sizes Redpanda is already writing with default settings. Then I also wanted to add a "series" of varying iodepth tests, which I sort of arbitrarily chose to be 16K one. I didn't want to do both to keep the number of tests down, and I think maybe I favored 16K over 4K in part because 4K already had parallelism=2, and I wanted 1 and didn't want to charge the existing 4K test to keep some continuity with old results.

That said, very open to changing it. What is your view on the ideal series of tests to run?

Reply:
> One thing to note is that the parallelism factor here is then multiplied by the shard count

Wait but right now this all happens on shard zero only. Are you saying we still multiply it by the shard count?

> That said, very open to changing it. What is your view on the ideal series of tests to run?

I don't feel strongly. Just really coming from the classic 4k test and I guess it matches the min amount we write.

I guess the 512Kib test is actually the least relevant one for RP as we never write sizes bigger than 16Kib (only when fetching from TS).

Reply:
> Wait but right now this all happens on shard zero only. Are you saying we still multiply it by the shard count?

No I was simply mistaken. I thought this ran on all shards, but as you say it seems to run only one shard. I was thrown off by [this code and comment](https://github.com/redpanda-data/redpanda/blob/75dbc2093b7cc9e86b81cfe5667b86315e227210/src/v/cluster/self_test_rpc_types.h#L61-L62). Perhaps vestigial?

So I will adjust the numbers to hit higher io depths, and maybe add 1 more test.

> Just really coming from the classic 4k test and I guess it matches the min amount we write.

I'll change it to 4K.

> I guess the 512Kib test is actually the least relevant one for RP as we never write sizes bigger than 16Kib (only when fetching from TS).

It's definitely the least useful for evaluating RP performance at the default settings. As a test to understand more about the disk, especially disks with characteristics different than the most common ones we run on I think it's fine because it is a "max throughput" test, and if it gets a much higher number than the other tests with small blocks then we've learned something.